From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Wed, 10 Jul 2019 19:22:48 +0200
Subject: librem5: Disable print

The GTK widget can't fit in the screen yet, so we better disable it for
now.
---
 src/ephy-lockdown.c   |  2 --
 src/ephy-window.c     | 12 +-----------
 src/window-commands.c | 17 -----------------
 src/window-commands.h |  3 ---
 4 files changed, 1 insertion(+), 33 deletions(-)

diff --git a/src/ephy-lockdown.c b/src/ephy-lockdown.c
index fe08141..13362ad 100644
--- a/src/ephy-lockdown.c
+++ b/src/ephy-lockdown.c
@@ -85,8 +85,6 @@ static const BindAction window_actions[] = {
 
   { EPHY_PREFS_LOCKDOWN_BOOKMARK_EDITING, "bookmark-page", "enabled" },
 
-  { EPHY_PREFS_LOCKDOWN_PRINTING, "print", "enabled" },
-
   { EPHY_PREFS_LOCKDOWN_SAVE_TO_DISK, "save-as", "enabled" }
 };
 
diff --git a/src/ephy-window.c b/src/ephy-window.c
index 46b592f..a4bd4cf 100644
--- a/src/ephy-window.c
+++ b/src/ephy-window.c
@@ -100,7 +100,6 @@ const struct {
   { "win.zoom-in", { "<Primary>plus", "<Primary>KP_Add", "<Primary>equal", "ZoomIn", NULL } },
   { "win.zoom-out", { "<Primary>minus", "<Primary>KP_Subtract", "ZoomOut", NULL } },
   { "win.zoom-normal", { "<Primary>0", "<Primary>KP_0", NULL } },
-  { "win.print", { "<Primary>P", NULL } },
   { "win.find", { "<Primary>F", NULL } },
   { "win.find-prev", { "<shift><Primary>G", NULL } },
   { "win.find-next", { "<Primary>G", NULL } },
@@ -470,14 +469,6 @@ sync_tab_load_status (EphyWebView     *view,
 
   loading = ephy_web_view_is_loading (view);
 
-  action_group = gtk_widget_get_action_group (GTK_WIDGET (window), "win");
-
-  /* disable print while loading, see bug #116344 */
-  action = g_action_map_lookup_action (G_ACTION_MAP (action_group),
-                                       "print");
-  ephy_action_change_sensitivity_flags (G_SIMPLE_ACTION (action),
-                                        SENS_FLAG_LOADING, loading);
-
   action_group = gtk_widget_get_action_group (GTK_WIDGET (window), "toolbar");
 
   action = g_action_map_lookup_action (G_ACTION_MAP (action_group),
@@ -829,7 +820,6 @@ static const GActionEntry window_entries [] = {
   { "zoom-in", window_cmd_zoom_in },
   { "zoom-out", window_cmd_zoom_out },
   { "zoom-normal", window_cmd_zoom_normal },
-  { "print", window_cmd_print },
   { "find", window_cmd_find },
   { "find-prev", window_cmd_find_prev },
   { "find-next", window_cmd_find_next },
@@ -1015,7 +1005,7 @@ _ephy_window_set_default_actions_sensitive (EphyWindow *window,
   int i;
 
   const char *action_group_actions[] = {
-    "save-as", "save-as-application", "print",
+    "save-as", "save-as-application",
     "find", "find-prev", "find-next",
     "bookmark-page", "encoding", "page-source",
     "send-to",
diff --git a/src/window-commands.c b/src/window-commands.c
index 4ed4184..b71321b 100644
--- a/src/window-commands.c
+++ b/src/window-commands.c
@@ -2271,23 +2271,6 @@ window_cmd_delete (GSimpleAction *action,
   }
 }
 
-void
-window_cmd_print (GSimpleAction *action,
-                  GVariant      *parameter,
-                  gpointer       user_data)
-{
-  EphyWindow *window = user_data;
-  EphyEmbed *embed;
-  EphyWebView *view;
-
-  embed = ephy_embed_container_get_active_child
-            (EPHY_EMBED_CONTAINER (window));
-  g_assert (EPHY_IS_EMBED (embed));
-  view = ephy_embed_get_web_view (embed);
-
-  ephy_web_view_print (view);
-}
-
 void
 window_cmd_find (GSimpleAction *action,
                  GVariant      *parameter,
diff --git a/src/window-commands.h b/src/window-commands.h
index 784363b..b96e98e 100644
--- a/src/window-commands.h
+++ b/src/window-commands.h
@@ -113,9 +113,6 @@ void window_cmd_paste_as_plain_text             (GSimpleAction *action,
 void window_cmd_delete                          (GSimpleAction *action,
                                                  GVariant      *parameter,
                                                  gpointer       user_data);
-void window_cmd_print                           (GSimpleAction *action,
-                                                 GVariant      *parameter,
-                                                 gpointer       user_data);
 void window_cmd_find                            (GSimpleAction *action,
                                                  GVariant      *parameter,
                                                  gpointer       user_data);
